stdlib/gc.ml

```diff
@@ -109,7 +109,7 @@ type alarm_rec = {active : alarm; f : unit -> unit}
 let rec call_alarm arec =
   if Atomic.get arec.active then begin
     let finally () = finalise call_alarm arec in
-    Fun.protect arec.f ~finally
+    Fun.protect ~finally arec.f
```

Comment:
Out of curiosity, why change this?

Reply:
Because dune is unable to define per-module compile flags, and it needs to compile some of the the other modules in the same library with `-no-labels`.

This forces the order of the arguments here to be the same as they appear in the interface.

Reply:
Is there a way to check that we're not reintroducing this problem later? There's no check in the compiler codebase that makes sure this property is preserved.

Reply:
I think that would involve adding something like `-strict-labels` to ensure that labelled arguments are not used commutatively, or did you have something else in mind? I think it would be much nicer to be heading towards a world where `-labels`/`-nolabels` is removed completely, rather than adding a new feature to support it?